### PR TITLE
Assignments/string functions

### PR DESCRIPTION
**Pull request by szetingchen**

### CODE REVIEW

File: assignments/string-functions/code.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+const findFnord = (s) => {
+  return s.search("fnord")
+}
+
+const stringContains = (s1, s2) => {
+  return s1.search(s2) != -1
```
Comment (gigamonkey) [inline assignments/string-functions/code.js:6]: `search` is actually overpowered and won't always work. For instance, try calling this function with the arguments `'foo * bar'` and `'*'` and then see if you can figure out (maybe by reading some docs on MDN) why it explodes.

File: assignments/string-functions/code.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const swapFrontAndBack = (s) => {
+  return s.substring(s.length / 2) + s.substring(0, s.length / 2)
+}
+
+const simplePigLatin = (s, num) => {
+  return s.substring(num) + s.substring(0, num) + 'ay'
+}
+
+const isAllUpperCase = (s) => {
+  return s.search(/[abcdefghijklmnopqrstuvwxyz]/) == -1
```
Comment (gigamonkey) [inline assignments/string-functions/code.js:22]: This is a semi-legitimate use of `search` but only handles ASCII. Can you think of a way to do this with only the methods I've taught you. (I haven't really taught you `search` I just let you use it in the pig latin assignment.)

File: assignments/string-functions/code.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+const sameIgnoringCase = (s1, s2) => {
+  return s1.toLowerCase() == s2.toLowerCase()
```
Comment (gigamonkey) [inline assignments/string-functions/code.js:26]: Use `===` not `==`. (Doesn't matter here but get in the habit.)

File: assignments/string-functions/code.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+const everyOther = (s) => {
+  return s[0] + s[2] + s[4]
+}
+
+const upDownLastCharacter = (s) => {
+  return s[s.length - 1].toUpperCase() + s[s.length - 1].toLowerCase()
+}
+
+const firstName = (s) => {
+  return s.substring(0, s.search(/[ ]/))
```
Comment (gigamonkey) [inline assignments/string-functions/code.js:50]: This works. But you are obviously forgetting about `indexOf`.

**Review by gigamonkey — APPROVED** [review]
Other than using `search` when you should have been using other methods that I've actually taught you (and not always using `search` correctly) this is very good. You can merge but you should do the `stringContains` experiment I suggested.

**Comment by szetingchen** [comment]
done. merged before changes.

**Pull request merged**